archinstall/lib/args.py

```diff
@@ -391,12 +414,58 @@ def _parse_config(self) -> dict[str, Any]:
 			creds_data = self._fetch_from_url(self._args.creds_url)
 
 		if creds_data is not None:
-			config.update(json.loads(creds_data))
+			json_data = self._process_creds_data(creds_data)
+			if json_data is not None:
+				config.update(json_data)
 
 		config = self._cleanup_config(config)
 
 		return config
 
+	def _process_creds_data(self, creds_data: str) -> dict[str, Any] | None:
+		if creds_data.startswith('$'):  # encrypted data
+			if self._args.creds_decryption_key is not None:
+				try:
+					password = Password(plaintext=self._args.creds_decryption_key)
```

Comment:
The password is initialized with `plaintext` and then is only ever used as plaintext.

Reply:
Correct, I wanted to keep it consistent to use the wrapper data class and now throw around native types. It will make it easier in the future to perform actions on it if necessary and also easier figure out where passwords are used based on the type

Reply:
The wrapper data class is for passwords that are encrypted in a different manner, none of its features are used. It is perfectly fine to use native types. Everything looks complete here. I was not aware that types are intended for improved searching.

Reply:
Fair point, I'll update it